react/DeviceCoordinates.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { useState, useEffect, useCallback } from 'react'
+import GET_ADDRESS_FROM_GEOCOORDINATES from './graphql/queries.graphql'
+import { useAddressContext } from 'vtex.address-context/AddressContext'
+import { ButtonPlain, Spinner, Tooltip, IconLocation } from 'vtex.styleguide'
+import { FormattedMessage } from 'react-intl'
+import { useLazyQuery } from 'react-apollo'
+
+enum State {
+  PROMPT,
+  GRANTED,
+  LOADING,
+  DENIED,
+}
+
+const DeviceCoordinates: StorefrontFunctionComponent<{}> = () => {
+  const { setAddress } = useAddressContext()
+  const [state, setState] = useState<State>(State.PROMPT)
+  const [getAddressFromGeocoordinates, { error, loading, data }] = useLazyQuery(
+    GET_ADDRESS_FROM_GEOCOORDINATES
+  )
+
+  if (data) {
+    setAddress(data.reverseGeocode)
+  }
+
+  if (error) {
+    console.warn(`error ${error.message}`)
+  }
+
+  const onGetCurrentPositionSuccess = useCallback(
+    ({ coords }: Position) => {
+      console.log('Success!')
+      getAddressFromGeocoordinates({
+        variables: {
+          lat: coords.latitude.toString(),
+          lng: coords.longitude.toString(),
+          apiKey: 'PUT API KEY HERE',
```

Comment:
Any ideas on this? This most likely will break running this in production, but it wouldn't be correct to place the real apiKey here.

Reply:
It varies with accounts, right? So you should receive it from props https://github.com/vtex/address-form/blob/c08be681af8a352e53d37a8e3bc0ef831fa01c7a/demo/src/App.js#L85

Reply:
Yes, I'll add it as a prop to the `address-context` component. But still, when you want to really test it, you should place the real apiKey in the code (after the change, it would be in the `Example.tsx` component).